crates/burn-tensor/src/tensor/data.rs

```diff
@@ -148,6 +169,10 @@ impl<const D: usize, E: Element> Data<E, D> {
         Data {
             value,
             shape: self.shape,
+            // Note: conversion of quantized values assumes that the target element type can represent
+            // the quantized values adequatly. It is implemented for good measure, but users
+            // should be mindful of the possible impacts.
+            quantization: self.quantization,
```

Comment:
Not entirely sure how we should handle conversion when quantization is applied. Just documentation? or warn? or panic?

Reply:
warn for now.

Reply:
When I proposed to have everything as bytes, it's because quantization data doesn't have a precision settings, so converting doesn't make sense. As of right now it seems like you can't have i32 Int tensors as well as quantized tensors :/